test/axes/axisTests.ts

```diff
@@ -0,0 +1,652 @@
+      let annotationRect = d3.select(this);
+      let bbox = this.getBBox();
+      let insideAnnotationArea = bbox.x >= 0 && bbox.x + bbox.width <= axis.width() &&
+        bbox.y >= 0 && bbox.y + bbox.height <= axis.height() - axis.margin();
+      if (insideAnnotationArea) {
+        assert.strictEqual(annotationRect.attr("visibility"), "visible", `rect ${i} inside margin area should be visible`);
+      } else {
+        assert.strictEqual(annotationRect.attr("visibility"), "hidden", `rect ${i} outside margin area should be not visible`);
+      }
+    });
+    axis.destroy();
+    svg.remove();
+  });
+
+  it("shows annotation circles regardless if rectangles are hidden", () => {
+    let scale = new Plottable.Scales.Linear();
+    let annotatedTicks = [50, 51, 150];
+    scale.domain([0, 300]);
+
+    let axis = new Plottable.Axis(scale, "bottom");
+    axis.annotatedTicks(annotatedTicks);
+    axis.annotationsEnabled(true);
+
+    let svg = TestMethods.generateSVG(300, 300);
+    axis.renderTo(svg);
+
+    let annotationCircles = axis.content().selectAll(`.${Plottable.Axis.ANNOTATION_CIRCLE_CLASS}`);
+
+    assert.strictEqual(annotationCircles.size(), annotatedTicks.length, "same number of annotation circles as ticks");
+
+    annotationCircles.each(function(d, i) {
+      let annotationCircle = d3.select(this);
+      assert.notStrictEqual(annotationCircle.attr("visibility"), "hidden", `circle ${i} inside margin area should be visible`);
+    });
+    axis.destroy();
+    svg.remove();
+  });
+
+  it("does not render the null as an annotation", () => {
+    let annotatedTicks: Date[] = [null];
+    let scale = new Plottable.Scales.Time();
+    scale.domain([new Date(1994, 11, 17), new Date(1995, 11, 17)]);
+    let axis = new Plottable.Axis(scale, "bottom");
+    axis.annotationsEnabled(true);
+    let svg = TestMethods.generateSVG(300, 300);
+    axis.renderTo(svg);
+
+    axis.annotatedTicks(annotatedTicks);
+
+    assert.strictEqual(axis.content().selectAll(`.${Plottable.Axis.ANNOTATION_LABEL_CLASS}`).size(), 0, "no annotated ticks");
+
+    axis.destroy();
+    svg.remove();
+  });
+
+  it("only renders unique annotated ticks", () => {
+    let annotatedTicks = [150, 150, 200];
+    let scale = new Plottable.Scales.Linear();
+    scale.domain([100, 200]);
+    let axis = new Plottable.Axis(scale, "bottom");
+    axis.annotationsEnabled(true);
+    let svg = TestMethods.generateSVG(300, 300);
+    axis.renderTo(svg);
+
+    axis.annotatedTicks(annotatedTicks);
+
+    let annotatedTickSet = new Plottable.Utils.Set();
+    annotatedTicks.forEach((annotatedTick) => {
+      annotatedTickSet.add(annotatedTick);
+    });
+    assert.strictEqual(axis.content().selectAll(`.${Plottable.Axis.ANNOTATION_LABEL_CLASS}`).size(),
+      annotatedTickSet.size, "only unique annotations rendered");
+
+    axis.destroy();
+    svg.remove();
+  });
+
+  it("renders a duplicate tick in the first tier it can fit in", () => {
```

Comment:
The previous test makes sure that only the unique ones render.  However there was an earlier bug where duplicating an annotated tick would offset the tick into a different row.

Ex. Earlier bug had having annotated ticks be [50, 50, 50, 50] render single annotation for 50 but in the 4th row.